### ISSUE TITLE
Does segmentation polygon annotation work with copy paste hyperparameter?

### ISSUE THREAD

**Issue by Evzen-T** [ultralytics/yolov5#10264]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I've been doing segmentation polygon annotation with label-studios, exporting it with coco format.
Afterwards i upload the coco format i got from label-studios into roboflow, and export it with yolov5 format.

However when using the copy paste hyperparameter, the train batches does not have copypasted images shown when i checked after training.
![train_batch0](https://user-images.githubusercontent.com/113011632/203464569-1b71759f-fc54-4923-8b95-970455907b97.jpg)

**Comment by Evzen-T** [comment]
Or does segmentation polygon annotation only works with roboflow?

**Comment by Laughing-q** [comment]
@Evzen-T copy-paste works with mosaic augmentation, but from your images seems you didn't use mosaic.
[https://github.com/ultralytics/yolov5/blob/7398d2d77cbac9f66259926d49c26bfa3c257a9b/utils/dataloaders.py#L792](https://github.com/ultralytics/yolov5/blob/7398d2d77cbac9f66259926d49c26bfa3c257a9b/utils/dataloaders.py#L792)

**Comment by Evzen-T** [comment]
Wait so thats not mosaic? That is my image i got from the train batch 0, with the mosaic hyperparameter on.

This are my hyperparameters:
lr0: 0.01  # initial learning rate (SGD=1E-2, Adam=1E-3)
lrf: 0.01  # final OneCycleLR learning rate (lr0 * lrf)
momentum: 0.937  # SGD momentum/Adam beta1
weight_decay: 0.0005  # optimizer weight decay 5e-4
warmup_epochs: 3.0  # warmup epochs (fractions ok)
warmup_momentum: 0.8  # warmup initial momentum
warmup_bias_lr: 0.1  # warmup initial bias lr
box: 0.05  # box loss gain
cls: 0.5  # cls loss gain
cls_pw: 1.0  # cls BCELoss positive_weight
obj: 1.0  # obj loss gain (scale with pixels)
obj_pw: 1.0  # obj BCELoss positive_weight
iou_t: 0.20  # IoU training threshold
anchor_t: 4.0  # anchor-multiple threshold
# anchors: 3  # anchors per output layer (0 to ignore)
fl_gamma: 0.0  # focal loss gamma (efficientDet default gamma=1.5)
hsv_h: 0.015  # image HSV-Hue augmentation (fraction)
hsv_s: 0.7  # image HSV-Saturation augmentation (fraction)
hsv_v: 0.4  # image HSV-Value augmentation (fraction)
degrees: 0.0  # image rotation (+/- deg)
translate: 0.1  # image translation (+/- fraction)
scale: 0.5  # image scale (+/- gain)
shear: 0.0  # image shear (+/- deg)
perspective: 0.0  # image perspective (+/- fraction), range 0-0.001
flipud: 0.0  # image flip up-down (probability)
fliplr: 0.0  # image flip left-right (probability)
mosaic: 1.0  # image mosaic (probability)
mixup: 0.0  # image mixup (probability)
copy_paste: 1.0  # segment copy-paste (probability)

**Comment by Laughing-q** [comment]
@Evzen-T if you're setting `augment=True` and `rect=False`, this `mosaic=1.0` in this hyp file should work.
[https://github.com/ultralytics/yolov5/blob/7398d2d77cbac9f66259926d49c26bfa3c257a9b/utils/dataloaders.py#L455](https://github.com/ultralytics/yolov5/blob/7398d2d77cbac9f66259926d49c26bfa3c257a9b/utils/dataloaders.py#L455)
you might want to debug here to see if mosaic is True.
[https://github.com/ultralytics/yolov5/blob/7398d2d77cbac9f66259926d49c26bfa3c257a9b/utils/dataloaders.py#L654](https://github.com/ultralytics/yolov5/blob/7398d2d77cbac9f66259926d49c26bfa3c257a9b/utils/dataloaders.py#L654)

**Comment by Laughing-q** [comment]
@Evzen-T actually the copy-paste is working with mosaic augmentation, they're bound. If you set `rect=True` then `self.mosaic=False` then mosaic and copy-paste both won't work. You have to make `self.mosaic=True` if you want to use copy-paste.

**Comment by Evzen-T** [comment]
Ok thanks for explaining it to me, i'll try it out

**Comment by Evzen-T** [comment]
Yup it worked out as you've said. Thanks alot @Laughing-q 
![image](https://user-images.githubusercontent.com/113011632/203498748-ea098105-7896-439c-8226-76b854b12421.png)

**Comment by ryouchinsa** [comment]
RectLabel is an offline image annotation tool for object detection and segmentation.
Although this is not an open source program, with RectLabel, you can label polygons and keypoints in the YOLO format.
https://rectlabel.com/help#xml_to_yolo

Please give us your feedback about the combination of YOLOv5 and RectLabel.
